File: Python/Simplify Path.py
class Solution:
    # @param path, a string
    # @return a string
    def simplifyPath(self, path):
        result = []
        pathList = path.split('/')
        for content in pathList:
            if content:
                if content == '..':
                    try:
                        result.pop()
                    except:
                        result = []
                elif content != '.':
                    result.append(content)
        return '/'+'/'.join(result)

File: Java/src/Simplify Path.java
import java.util.Deque;
import java.util.LinkedList;

public class Solution
{
  public String simplifyPath(String path)
  {
    Deque<String> deque = new LinkedList<String>();
    String[] pathArray = path.split("/");
    for (String s : pathArray)
    {
      if (s.equals(".."))
      {
        if (!deque.isEmpty())
        {
          deque.pop();
        }
      }

      else if (!s.equals(".") && !s.equals(""))
      {
        deque.push(s);
      }
    }

    String ret = "";
    for (String s : deque)
    {
      ret = "/" + s + ret;
    }
    return ret.isEmpty() ? "/" : ret;
  }
}
